Electron37 cloud build

caneraltinbasak opened the pull request:
This pull request introduces major improvements to the GitHub Actions CI/CD workflows for the meta-browser project. The changes add support for running builds and tests on AWS EC2 instances for both Chromium and Electron, enable matrix builds across multiple configurations, and provide a new workflow for cleaning up EFS cache storage. The updates make the CI system more scalable, configurable, and better suited for large builds.

**New EC2-based CI workflow and matrix build support:**

* Added a new workflow file `bs_meta_browser_ci_ec2.yml` that provisions EC2 runners, tags resources, and orchestrates build/test jobs for meta-browser on AWS, including automatic runner startup and cleanup.
* Updated `chromium.yml` and `electron.yml` to support matrix builds on AWS EC2, with configurable instance types, build parameters, and the option to fall back to local runners for manual dispatch. This enables parallel builds across Yocto versions, platforms, architectures, and libc flavors. [[1]](diffhunk://#diff-8f5503c36aa7e90a80695b902b7758301560c1c8c3e8e961455dbddad3a7b0f2R14-R28) [[2]](diffhunk://#diff-8f5503c36aa7e90a80695b902b7758301560c1c8c3e8e961455dbddad3a7b0f2R42-R72) [[3]](diffhunk://#diff-e3f2d8bfe2b2f0cc1deb963d1584f4f7cf6fff5111a0fd4c1d90904e1d7c93ffR14-R28) [[4]](diffhunk://#diff-e3f2d8bfe2b2f0cc1deb963d1584f4f7cf6fff5111a0fd4c1d90904e1d7c93ffR42-R72)

**Infrastructure and configuration enhancements:**

* Added OIDC authentication permissions (`id-token: write`) to workflows for secure AWS credential management. [[1]](diffhunk://#diff-8f5503c36aa7e90a80695b902b7758301560c1c8c3e8e961455dbddad3a7b0f2R42-R72) [[2]](diffhunk://#diff-e3f2d8bfe2b2f0cc1deb963d1584f4f7cf6fff5111a0fd4c1d90904e1d7c93ffR42-R72)
* Improved EC2 runner tagging for better traceability and resource management, including passing build metadata as AWS resource tags.

**EFS cache management:**

* Introduced a new workflow `cleanup-efs-cache.yml` to automate cleanup of Yocto downloads and sstate-cache directories on EFS using a temporary EC2 runner, with options for selective cleanup and disk usage reporting.

**Minor fixes and consistency:**

* Updated test repo cloning in build scripts to remove hardcoded branch, ensuring consistency for both Chromium and Electron workflows. [[1]](diffhunk://#diff-8f5503c36aa7e90a80695b902b7758301560c1c8c3e8e961455dbddad3a7b0f2L60-R101) [[2]](diffhunk://#diff-e3f2d8bfe2b2f0cc1deb963d1584f4f7cf6fff5111a0fd4c1d90904e1d7c93ffL60-R101)

caneraltinbasak committed:
electron cloud build

caneraltinbasak committed:
Use aws by default, fix dependencies

caneraltinbasak committed:
efs dns debug

caneraltinbasak committed:
more debug

caneraltinbasak committed:
Fix EFS mounting issues

- Add NFS egress rule (port 2049) to VPC security group in Terraform
- Improve EFS mounting logic with DNS and IP fallback
- Add better debugging and status indicators
- Fix Yocto build environment setup for meta-chromium-test script
- Install kas tool and set up proper directory structure
- Use electron-master branch for chromium builds

caneraltinbasak committed:
Fix YAML syntax and improve EFS mounting

- Fix corrupted YAML structure in build workflow
- Add kas installation for Yocto builds
- Improve EFS mounting with IP fallback and better error handling
- Add proper directory structure setup for build script
- Enhanced debugging and status reporting
- Use electron-master branch for electron builds

caneraltinbasak committed:
fix build.sh parameter order

caneraltinbasak committed:
Fix electron build branch

caneraltinbasak committed:
Fix Chromium/Electron builds.

caneraltinbasak committed:
Add a new action for cleaning up efs

caneraltinbasak committed:
Fix for premature job cancellation

caneraltinbasak committed:
Do not fail-fast

Continue running jobs even if one fails

caneraltinbasak committed:
use larger aws instances

caneraltinbasak committed:
Use smaller AWS instances with swap.

caneraltinbasak committed:
Enhance artifact upload with 128GB swap and better S3 diagnostics

- Increase swap from 4GB to 128GB on NVMe SSD for memory-intensive Yocto builds
- Add swap performance optimization (swappiness=10, vfs_cache_pressure=1)
- Enhance S3 upload with detailed logging and error handling
- Add pre-upload diagnostics (disk space, upload size, S3 access test)
- Remove --quiet flag for better error visibility
- Add post-upload verification and file listing
- Improve timeout settings for reliable uploads

caneraltinbasak committed:
refresh aws credentials.

caneraltinbasak committed:
use unique name for aws ec2 instances

caneraltinbasak committed:
increase timeout

caneraltinbasak committed:
increase swap size

caneraltinbasak committed:
Use updated AMI image and do not install new packages

caneraltinbasak committed:
Increase AWS session duration and use newer plugin

caneraltinbasak committed:
fix sstate cache paths

caneraltinbasak committed:
Super unique name

caneraltinbasak committed:
Do not use ccache and build on nvme, rather than rootfs

caneraltinbasak committed:
Fix permission issues

caneraltinbasak committed:
refactor and fix permission problems

caneraltinbasak committed:
fix permission errors

caneraltinbasak committed:
Fix permissions

caneraltinbasak commented:
Build succeeded. I'll squash and merge.

caneraltinbasak committed:
Remove workspace file

t-bashir-bs commented on .github/workflows/cleanup-efs-cache.yml:
```diff
@@ -0,0 +1,169 @@
+          github-token: ${{ secrets.GH_PERSONAL_ACCESS_TOKEN }}
+          ec2-image-id: ami-0ba9883b710398fbb
+          ec2-instance-type: ${{ inputs.instance_type }}
+          subnet-id: subnet-06b2fe35df970cb71
```
Is this different to ${{ env.VPC_SUBNET_ID }}?

caneraltinbasak replied:
done

t-bashir-bs commented on .github/workflows/cleanup-efs-cache.yml:
```diff
@@ -0,0 +1,169 @@
+          ec2-image-id: ami-0ba9883b710398fbb
+          ec2-instance-type: ${{ inputs.instance_type }}
+          subnet-id: subnet-06b2fe35df970cb71
+          security-group-id: sg-0905fa8cd994895e8
```
Is this different to ${{ env.VPC_SG_ID }}?

caneraltinbasak replied:
done

t-bashir-bs commented on .github/workflows/cleanup-efs-cache.yml:
```diff
@@ -0,0 +1,169 @@
+        uses: aws-actions/configure-aws-credentials@v4
+        with:
+          aws-region: ${{ inputs.aws_region }}
+          role-to-assume: "arn:aws:iam::195607249165:role/github-actions-meta-browser-repo"
```
Should this be hard coded?

caneraltinbasak replied:
done

t-bashir-bs commented on .github/workflows/cleanup-efs-cache.yml:
```diff
@@ -0,0 +1,169 @@
+        uses: aws-actions/configure-aws-credentials@v4
+        with:
+          aws-region: ${{ inputs.aws_region }}
+          role-to-assume: "arn:aws:iam::195607249165:role/github-actions-meta-browser-repo"
```
Should this be hard coded?

caneraltinbasak replied:
done

t-bashir-bs commented on .github/workflows/chromium.yml:
```diff
@@ -24,11 +29,37 @@ permissions:
+      chromium_version: ${{ matrix.chromium_version }}
+      libc_flavour: ${{ matrix.libc_flavour }}
+      arch: ${{ matrix.arch }}
+      aws_arn_role: "arn:aws:iam::195607249165:role/github-actions-meta-browser-repo"
```
Should this be hard coded?

caneraltinbasak replied:
done

t-bashir-bs commented on .github/workflows/bs_meta_browser_build_and_test.yml:
```diff
@@ -0,0 +1,514 @@
+            
+            # Verify AWS credentials are working
+            echo "Verifying AWS credentials..."
+            aws sts get-caller-identity
```
The comment says verifying credentials, but we don't do anything if there is a failure?

caneraltinbasak replied:
done

t-bashir-bs commented on .github/workflows/bs_meta_browser_build_and_test.yml:
```diff
@@ -0,0 +1,514 @@
+            
+            echo "Starting S3 upload..."
+            # Use verbose mode and enable progress tracking
+            if aws s3 sync "$DEPLOY_DIR/" "$S3_PATH" --cli-read-timeout 0 --cli-connect-timeout 60; then
```
For the Electron CI I added --quiet and didn't use verbose mode for large copies as I think it was causing a lot of tracing spam in the output. Is the amount of messages reasonable in verbose mode?

caneraltinbasak replied:
done

t-bashir-bs commented on .github/workflows/bs_meta_browser_ci_ec2.yml:
```diff
@@ -0,0 +1,156 @@
+        required: true
+        type: string
+
+      instance_type:
```
In the Electron CI scripts I used a choice to ensure we don't just set an inappropriate instance type...

```
      instance_type:
        description: 'EC2 Instance Type'
        type: choice
        options:
        - c6a.large
        - c6a.xlarge
        - c6a.2xlarge
        - c6a.4xlarge
        - c6a.8xlarge
```

caneraltinbasak replied:
done

t-bashir-bs commented on .github/workflows/cleanup-efs-cache.yml:
```diff
@@ -0,0 +1,169 @@
+        type: string
+        default: 'us-east-1'
+      
+      instance_type:
```
See previous comment about using a choice instead of a string here.

caneraltinbasak replied:
This is just a cleanup. Doesn't matter what the instance is.

t-bashir-bs commented on .github/workflows/bs_meta_browser_build_and_test.yml:
```diff
@@ -0,0 +1,514 @@
+          
+          # Check if we can reach AWS services
+          echo "=== AWS Connectivity Check ==="
+          curl -s --connect-timeout 10 https://s3.amazonaws.com && echo "S3 reachable" || echo "S3 not reachable"
```
Should we just echo "S3 not reachable" or exit 1?

caneraltinbasak replied:
done

caneraltinbasak committed:
Address github copilot review comments

caneraltinbasak committed:
Address Tariq's review comments.

caneraltinbasak committed:
Use unique session names and labels

caneraltinbasak committed:
Do not recursively change permissions

caneraltinbasak committed:
shorten role names

caneraltinbasak committed:
Use main branch of meta-chromium-test

During development, I've used electron-master branch. electron-master
is now merged to main. Switch to that branch
